### PR TITLE
Add compute engine system tests

### LINKED ISSUE #40: Create system tests for GCE credentials

### PR DESCRIPTION
**Pull request by jonparrott**
Resolves #40

### CODE REVIEW

File: google/auth/compute_engine/credentials.py [hunk 1]
```diff
@@ -71,7 +70,7 @@ def _retrieve_info(self, request):
             service_account=self._service_account_email)
 
         self._service_account_email = info['email']
-        self._scopes = _helpers.string_to_scopes(info['scopes'])
+        self._scopes = info['scopes']
```
Comment (dhermes) [inline google/auth/compute_engine/credentials.py]: Not needed anymore?

File: system_tests/test_compute_engine.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.auth import _helpers
+from google.auth import compute_engine
+from google.auth.compute_engine import _metadata
+import pytest
```
Comment (dhermes) [inline system_tests/test_compute_engine.py]: 3rd-party imports come before local imports?

File: system_tests/test_compute_engine.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+
+@pytest.fixture(autouse=True)
+def check_gce_environment(request):
+    if not _metadata.ping(request):
+        pytest.skip('Compute Engine metadata service is not available.')
+
+
```
Comment (dhermes) [inline system_tests/test_compute_engine.py]: Missing test: ADC correctly spits out an instance of `compute_engine.Credentials`

File: system_tests/test_compute_engine.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+def test_refresh(request, token_info):
```
Comment (dhermes) [inline system_tests/test_compute_engine.py]: I'm like "where does the `token_info` fixture come from?". Would be real grok if I didn't remember you merged it yesterday.

File: system_tests/test_compute_engine.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    credentials = compute_engine.Credentials()
+
+    credentials.refresh(request)
+
+    assert credentials.token is not None
+    assert credentials._service_account_email is not None
```
Comment (dhermes) [inline system_tests/test_compute_engine.py]: Is there a public property to access `service_account_email`?

File: system_tests/test_compute_engine.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+    info = token_info(credentials.token)
+    info_scopes = _helpers.string_to_scopes(info['scope'])
```
Comment (dhermes) [inline system_tests/test_compute_engine.py]: :+1:

File: system_tests/test_service_account.py [hunk 7]
```diff
@@ -38,6 +39,7 @@ def test_refresh_success(request, credentials, token_info):
     info = token_info(credentials.token)
 
     assert info['email'] == credentials._service_account_email
-    assert info['scope'] == (
-        'https://www.googleapis.com/auth/userinfo.email '
-        'https://www.googleapis.com/auth/userinfo.profile')
+    info_scopes = _helpers.string_to_scopes(info['scope'])
```
Comment (dhermes) [inline system_tests/test_service_account.py]: :+1:

File: google/auth/compute_engine/credentials.py (hunk 1 above)
Comment (jonparrott) [inline google/auth/compute_engine/credentials.py]: Yeah, the info endpoint returns the scopes as a list, not a string like tokeninfo/oauth2.

File: system_tests/test_compute_engine.py (hunk 2 above)
Comment (jonparrott) [inline system_tests/test_compute_engine.py]: Fair enough.

File: system_tests/test_compute_engine.py (hunk 3 above)
Comment (jonparrott) [inline system_tests/test_compute_engine.py]: See #55

File: system_tests/test_compute_engine.py (hunk 4 above)
Comment (jonparrott) [inline system_tests/test_compute_engine.py]: Sorry, I need it in multiple tests and the best place to put it is in `conftest.py`.

File: system_tests/test_compute_engine.py (hunk 5 above)
Comment (jonparrott) [inline system_tests/test_compute_engine.py]: Nope.

**Comment by dhermes** [comment]
LGTM (in light of the existence of #55).

Also, don't block here, but as a general rule I try to have system tests only rely on public attributes. Why isn't it valuable for the svc. acct. email to be public?

**Comment by jonparrott** [comment]
> Why isn't it valuable for the svc. acct. email to be public?

It could be useful, but I want to err on the side of being limited in our public interface.

**Pull request merged**